src/avram/errors.cr

```diff
@@ -59,4 +59,22 @@ module Avram
       super message
     end
   end
+
+  class ConnectionError < AvramError
+    def initialize(connection_details : URI)
+      error = String.build do |message|
+        message << "Failed to connect to databse '#{connection_details.path.try(&.[1..-1])}' with username #{connection_details.user}.\n"
+        message << "You can try these steps to help:"
+        message << '\n'
+        message << '\n'
+        message << "  ▸ Ensure you have the correct settings in config/database.cr\n"
```

Comment:
Nice. We can add some comments here in the generated code to help people find a suitable username/password for command Postgres installations

Reply:
People don't read (myself included) so maybe we can shorten this to `Check connection settings in 'config/databse.cr'`